In [1]:
import pandas

#pandas.options.display.max_columns = None
#pandas.options.display.max_rows = None

netflix = pandas.read_csv("netflixData.csv")
netflix

Show Id                          Title  \
0     cc1b6ed9-cf9e-4057-8303-34577fb54477                       (Un)Well   
1     e2ef4e91-fb25-42ab-b485-be8e3b23dedb                         #Alive   
2     b01b73b7-81f6-47a7-86d8-acb63080d525  #AnneFrank - Parallel Stories   
3     b6611af0-f53c-4a08-9ffa-9716dc57eb9c                       #blackAF   
4     7f2d4170-bab8-4d75-adc2-197f7124c070               #cats_the_mewvie   
...                                    ...                            ...   
5962  62b8b682-f191-4c10-aa04-32319329bd8d                      الف مبروك   
5963  5bed77ab-5e31-4216-8b51-44c9a35442e6                   دفعة القاهرة   
5964  4661ec0c-8692-4661-bc76-a96412b311fd                           海的儿子   
5965  145c93a7-1924-403c-a933-4ede8ad66f26                        반드시 잡는다   
5966  d4613d34-cb71-4bd6-b570-fa857f02c44d            최강전사 미니특공대 : 영웅의 탄생   

                                            Description  \
0     This docuseries takes a deep dive into the luc...   
1     As a grisly virus rampages a city, a lone man ...   
2     Through her diary, Anne Frank's story is retol...   
3     Kenya Barris and his family navigate relations...   
4     This pawesome documentary explores how our fel...   
...                                                 ...   
5962  On his wedding day, an arrogant, greedy accoun...   
5963  A group of women leaves Kuwait to attend unive...   
5964  Two brothers start a new life in Singapore, wh...   
5965  After people in his town start turning up dead...   
5966  Miniforce, a special task force of elite range...   

                         Director  \
0                             NaN   
1                          Cho Il   
2     Sabina Fedeli, Anna Migotto   
3                             NaN   
4                Michael Margolis   
...                           ...   
5962            Ahmed Nader Galal   
5963                          NaN   
5964                          NaN   
5965                Hong-seon Kim   
5966                Young Jun Lee   

                                              Genres  \
0                                         Reality TV   
1     Horror Movies, International Movies, Thrillers   
2                Documentaries, International Movies   
3                                        TV Comedies   
4                Documentaries, International Movies   
...                                              ...   
5962          Comedies, Dramas, International Movies   
5963               International TV Shows, TV Dramas   
5964               International TV Shows, TV Dramas   
5965         Dramas, International Movies, Thrillers   
5966                        Children & Family Movies   

                                                   Cast Production Country  \
0                                                   NaN      United States   
1                              Yoo Ah-in, Park Shin-hye        South Korea   
2                           Helen Mirren, Gengher Gatti              Italy   
3     Kenya Barris, Rashida Jones, Iman Benson, Genn...      United States   
4                                                   NaN             Canada   
...                                                 ...                ...   
5962  Ahmed Helmy, Laila Ezz El Arab, Mahmoud El Fis...              Egypt   
5963  Bashar al-Shatti, Fatima Al Safi, Maram Baloch...                NaN   
5964  Li Nanxing, Christopher Lee, Jesseca Liu, Appl...                NaN   
5965                                      Baek Yoon-sik        South Korea   
5966  Um Sang-hyun, Yang Jeong-hwa, Jeon Tae-yeol, S...                NaN   

      Release Date Rating  Duration Imdb Score Content Type         Date Added  
0           2020.0  TV-MA  1 Season     6.6/10      TV Show                NaN  
1           2020.0  TV-MA    99 min     6.2/10        Movie  September 8, 2020  
2           2019.0  TV-14    95 min     6.4/10        Movie       July 1, 2020  
3           2020.0  TV

In [23]:
# Which streaming platform has the most shows?
netflix = tvshow[tvshow["Netflix"] == 1]
nshows = len(netflix)

hulu = tvshow[tvshow["Hulu"] == 1]
hshows = len(hulu)

primevideo = tvshow[tvshow["Prime Video"] == 1]
pshows = len(primevideo)

disney = tvshow[tvshow["Disney+"] == 1]
dshows = len(disney)

print("Netflix show count:", nshows,"\nHulu show count:", 
      hshows, "\nPrime Video show count:", pshows, "\nDisney+ show count:", dshows)

Netflix show count: 1971 
Hulu show count: 1621 
Prime Video show count: 1831 
Disney+ show count: 351


# Here I was finding which streaming platform had the most tv shows across Netflix, Hulu, Prime Video, and Disney+. My findings show that Netflix had the most tv shows totalling at 1,971 with Prime Video at a close second with 1,831 shows on their platform. That being said Disney+ had the least amount of shows at 351.


In [26]:
# What is the average tv show year (year the tv show was produced) for each streaming platform?
netflix = tvshow[tvshow["Netflix"] == 1]
navg_year = netflix["Year"].mean()

hulu = tvshow[tvshow["Hulu"] == 1]
havg_year = hulu["Year"].mean()

primevideo = tvshow[tvshow["Prime Video"] == 1]
pavg_year = primevideo["Year"].mean()

disney = tvshow[tvshow["Disney+"] == 1]
davg_year = disney["Year"].mean()

print("Netflix's Average Show Year:", navg_year,"\nHulu's Average Show Year:", 
      havg_year, "\nPrime Video's Average Show Year:", pavg_year, "\nDisney+'s Average Show Year:", davg_year)

Netflix's Average Show Year: 2016.2836123795028 
Hulu's Average Show Year: 2010.4947563232572 
Prime Video's Average Show Year: 2009.773347897324 
Disney+'s Average Show Year: 2012.2421652421651


# Based on my findings Netflix had the most recent shows with an average show year of 2016. This is followed by Disney+ with an average show year of 2012. Lastly, Prime Video with the oldest average show years of 2009. These could also reflect the show’s likelihood to be on a platform based on the year they were produced.


In [25]:
# Which streaming platform caters more for kids age 7+?
netflix = tvshow[tvshow["Netflix"] == 1]
netflix_kid_age = netflix["Age"] == "7+"
netflix_showcount = 0
for age in netflix_kid_age:
    if age == True:
        netflix_showcount += 1

hulu = tvshow[tvshow["Hulu"] == 1]
hulu_kid_age = hulu["Age"] == "7+"
hulu_showcount = 0
for age in hulu_kid_age:
    if age == True:
        hulu_showcount += 1

primevideo = tvshow[tvshow["Prime Video"] == 1]
prime_video_kid_age = primevideo["Age"] == "7+"
prime_showcount = 0
for age in prime_video_kid_age:
    if age == True:
        prime_showcount += 1

disney = tvshow[tvshow["Disney+"] == 1]
disney_kid_age = disney["Age"] == "7+"
disney_showcount = 0
for age in disney_kid_age:
    if age == True:
        disney_showcount += 1

print("Netflix's kid friendy show count:", netflix_showcount,"\nHulu's kid friendy show count:", hulu_showcount,
      "\nPrime Video's kid friendy show count:", prime_showcount, "\nDisney+'s kid friendy show count:", disney_showcount)

Netflix's kid friendy show count: 299 
Hulu's kid friendy show count: 325 
Prime Video's kid friendy show count: 213 
Disney+'s kid friendy show count: 109


# Through my analysis I found Hulu to have the most kid friendly shows at 325. Here kid friendly is defined as shows with an age 7+ rating. Following Hulu we have Netflix with 299 shows for kids. Lastly, Disney+ has the lowest show count for kids at 109. 

In [24]:
# Which streaming platform has the most adult tv shows 18+?
netflix = tvshow[tvshow["Netflix"] == 1]
netflix_adult_age = netflix["Age"] == "18+"
netflix_showcount = 0
for age in netflix_adult_age:
    if age == True:
        netflix_showcount += 1

hulu = tvshow[tvshow["Hulu"] == 1]
hulu_adult_age = hulu["Age"] == "18+"
hulu_showcount = 0
for age in hulu_adult_age:
    if age == True:
        hulu_showcount += 1

primevideo = tvshow[tvshow["Prime Video"] == 1]
prime_video_adult_age = primevideo["Age"] == "18+"
prime_showcount = 0
for age in prime_video_adult_age:
    if age == True:
        prime_showcount += 1

disney = tvshow[tvshow["Disney+"] == 1]
disney_adult_age = disney["Age"] == "18+"
disney_showcount = 0
for age in disney_adult_age:
    if age == True:
        disney_showcount += 1

print("Netflix's adult show count:", netflix_showcount,"\nHulu's adult show count:", hulu_showcount,
      "\nPrime Video's adult show count:", prime_showcount, "\nDisney+'s adult show count:", disney_showcount)

Netflix's adult show count: 482 
Hulu's adult show count: 261 
Prime Video's adult show count: 164 
Disney+'s adult show count: 2


# Through my findings I found that Netflix had the highest show count for adults at 482 shows. Here adult meant people ages 18+. Following Netflix, Hulu had 261 shows with an adult rating. Lastly, Disney+ had only 2 18+ rating shows. This reflects Netflix's catering towards adults and with a big gap between its competitors. 